types/tizen-common-web/index.d.ts

```diff
@@ -1,84 +1,14699 @@
+}
+/**
+ * This callback interface specifies methods that are invoked when a package is installed, updated, or uninstalled.
+ */
+export interface PackageInformationEventCallback {
+    /**
+     * Called when a package is installed.
+     *
+     * @param info The information of the installed package.
+     */
+    oninstalled(info: PackageInformation): void;
+    /**
+     * Called when a package is uninstalled.
+     *
+     * @param id The ID of the uninstalled package.
+     */
+    onuninstalled(id: PackageId): void;
+    /**
+     * Called when a package is updated.
+     *
+     * @param info The information of the updated package.
+     */
+    onupdated(info: PackageInformation): void;
+}
+/**
+ * This callback interface specifies subscriptions for any notification on the progress or completion of requests.
+ */
+export interface PackageProgressCallback {
+    /**
+     * Called when the request is completed.
+     *
+     * @param id The package ID.
+     */
+    oncomplete(id: PackageId): void;
+    /**
+     * Called while the request is in progress.
+     *
+     * @param id The package ID.
+     * @param progress The progress in percentage.
+     */
+    onprogress(id: PackageId, progress: number): void;
+}
+/**
+ * The RequestCallback interface that defines the success method to be invoked when a client request is received.
+ *
+ * @remark Example of using can be find at [setRequestListener](iotcon.html#Resource::setRequestListener) code example.
+ */
+export interface RequestCallback {
+    /**
+     * Called when DELETE request was received.
+     *
+     * @param request That is request from server side.
+     */
+    ondelete(request: Request): void;
+    /**
+     * Called when GET request was received.
+     *
+     * @param request That is request from server side.
+     */
+    onget(request: Request): void;
+    /**
+     * Called when OBSERVE request was received.
+     *
+     * @param request That is request from server side.
+     * @param observeType The observation type of the request.
+     * @param observeId The observation id of the request.
+     */
+    onobserving(request: Request, observeType: ObserveType, observeId: number): void;
+    /**
+     * Called when POST request was received.
+     *
+     * @param request That is request from server side.
+     */
+    onpost(request: Request): void;
+    /**
+     * Called when PUT request was received.
+     *
+     * @param request That is request from server side.
+     */
+    onput(request: Request): void;
+}
+/**
+ * This is a common interface used by different types of
+ * object filters.
+ *
+ * Never use this base interface directly, instead use _AbstractFilter_ subtypes,
+ * such as _AttributeFilter_, _AttributeRangeFilter_, and _CompositeFilter_.
+ */
+export class AbstractFilter { // tslint:disable-line:no-unnecessary-class
+    constructor();
+}
+export interface AbstractFilter {
```

Comment:
This isn't the correct way to declare this. The `AbstractFilter` interface merges with the `AbstractFilter` class and represents an *instance* of the class, not the constructor side. The way you have it defined make it seem like `new (new AbstractFilter)()` would be valid.